### PR TITLE
Remove wcarthage workaround script and use xcframework dependencies

### PR DESCRIPTION
**Pull request by dersim-davaod**
### Description

Resolves: https://github.com/mapbox/navigation-sdks/issues/831

### CODE REVIEW

**Comment by 1ec5** [comment]
[Xcode 11 builds are failing](https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-ios/2489/workflows/5d44d531-5ff0-490a-9cce-9112c6c32de4/jobs/26479):

```
error: There is no XCFramework found at '/Users/distiller/project/Carthage/Build/MapboxDirections.xcframework'. (in target 'MapboxCoreNavigation' from project 'MapboxNavigation')
error: There is no XCFramework found at '/Users/distiller/project/Carthage/Build/MapboxMobileEvents.xcframework'. (in target 'MapboxCoreNavigation' from project 'MapboxNavigation')
error: There is no XCFramework found at '/Users/distiller/project/Carthage/Build/Polyline.xcframework'. (in target 'MapboxCoreNavigation' from project 'MapboxNavigation')
error: There is no XCFramework found at '/Users/distiller/project/Carthage/Build/Turf.xcframework'. (in target 'MapboxCoreNavigation' from project 'MapboxNavigation')
```

So are [Xcode 12.1 builds](https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-ios/2489/workflows/5d44d531-5ff0-490a-9cce-9112c6c32de4/jobs/26475):

```
	There is no XCFramework found at '/Users/distiller/project/Carthage/Build/MapboxDirections.xcframework'.
	There is no XCFramework found at '/Users/distiller/project/Carthage/Build/MapboxMobileEvents.xcframework'.
	There is no XCFramework found at '/Users/distiller/project/Carthage/Build/Polyline.xcframework'.
	There is no XCFramework found at '/Users/distiller/project/Carthage/Build/Turf.xcframework'.
	There is no XCFramework found at '/Users/distiller/project/Carthage/Build/MapboxSpeech.xcframework'.
	There is no XCFramework found at '/Users/distiller/project/Carthage/Build/Solar.xcframework'.
	There is no XCFramework found at '/Users/distiller/project/Carthage/Build/MapboxGeocoder.xcframework'.
	Testing cancelled because the build failed.
```

Are the new XCFramework references added to all the necessary targets within the Xcode project?

File: README.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ To install Mapbox Navigation using [CocoaPods](https://cocoapods.org/):
 
 ### Using Carthage
 
-To install Mapbox Navigation using [Carthage](https://github.com/Carthage/Carthage/) v0.35 or above:
+To install Mapbox Navigation using [Carthage](https://github.com/Carthage/Carthage/) v0.37.0 or above:
```
Comment (1ec5) [inline README.md:58]: 👌 Also update this requirement in custom-navigation.md.

File: README.md [hunk 2]
```diff
@@ -78,7 +78,7 @@ To install Mapbox Navigation using [Carthage](https://github.com/Carthage/Cartha
    github "mapbox/mapbox-navigation-ios" "v1.2.0-rc.1"
    ```
 
-1. Run `./Carthage/Checkouts/mapbox-navigation-ios/scripts/wcarthage.sh bootstrap --platform iOS --cache-builds --use-netrc`. (wcarthage.sh is a temporary replacement for `carthage` to work around [a linker error in Xcode 12](https://github.com/Carthage/Carthage/issues/3019).)
+1. Run `carthage bootstrap --platform ios --use-xcframeworks --cache-builds --use-netrc`. 
 
 1. Follow the rest of [Carthage’s iOS integration instructions](https://github.com/Carthage/Carthage#if-youre-building-for-ios-tvos-or-watchos). Your application target’s Embed Frameworks build phase should include `MapboxNavigation.framework`, `MapboxCoreNavigation.framework`, `MapboxNavigationNative.framework`, `MapboxCommon.framework`, and `MapboxAccounts.framework`.
```
Comment (1ec5) [inline README.md:83]: This should say .xcframework instead of .framework.

File: scripts/wcarthage.sh [hunk 3]
```diff
@@ -1,19 +0,0 @@
-#!/usr/bin/env bash
-
-# BASED ON THIS COMMENT: https://github.com/Carthage/Carthage/issues/3019#issuecomment-665136323
-
-# wcarthage.sh
-# Usage example: ./wcarthage.sh update --platform ios
```
Comment (1ec5) [inline scripts/wcarthage.sh:6]: We’ve been pointing developers to this script for a while now. These changes would most likely be going into a minor release, so for backwards compatibility, we should keep the script around. Let’s verify that it’s still possible to bootstrap an application target using this script and the accompanying .framework-based instructions, even as we change the installation instructions to recommend .xcframework going forward.

**Comment by S2Ler** [comment]
This PR work improved upon in #3031 and now can be closed. Thanks for contribution!

**Pull request closed without merging**